server/server/src/server.ts

```diff
@@ -0,0 +1,143 @@
+//
+// Copyright © 2021 Anticrm Platform Contributors.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import { createServer, IncomingMessage } from 'http'
+import WebSocket, { AddressInfo, Server } from 'ws'
+import { Storage, Tx } from '@anticrm/core'
+
+export interface ServerProtocol {
+  shutdown: () => void
+  address: () => { host: string; port: number }
+}
+export interface StorageProvider {
+  connect: (clientId: string, tx: (tx: Tx) => void) => Storage
+  close: (clientId: string) => void
+}
+
+export async function start (
+  host: string | undefined,
+  port: number,
+  provider: StorageProvider
+): Promise<ServerProtocol> {
+  console.log(`starting server on port ${port}...`)
+  console.log(`host: ${host ?? 'localhost'}`)
+
+  const server = createServer()
+  const wss = new Server({ noServer: true })
```

Comment:
What is this `noServer`? Seems does not used anywhere...

Reply:
I'm not sure but it not working without this option, I've could not find a documentation for it or why it is required. But all docs for ws also shows to pass it: https://github.com/websockets/ws/blob/master/README.md

Reply:
OK :)